worlds/stardew_valley/logic/tool_logic.py

```diff
@@ -40,13 +39,16 @@ def __init__(self, *args, **kwargs):
 class ToolLogic(BaseLogic[Union[ToolLogicMixin, HasLogicMixin, ReceivedLogicMixin, RegionLogicMixin, SeasonLogicMixin, MoneyLogicMixin, MagicLogicMixin]]):
     # Should be cached
     def has_tool(self, tool: str, material: str = ToolMaterial.basic) -> StardewRule:
+        if tool == Tool.fishing_rod:

```

Comment:
Honestly, I'd prefer an assert (or, even, nothing) here, and assuming that people simply don't call this method with the fishing rod, since it's not designed for it. I'm not a fan of adding an if for a bad usage of the method

Reply:
Having nothing to protect this function already proved that bugs will appear. I will go with the assert route.